functions/New-DbaFirewallRule.ps1

```diff
@@ -0,0 +1,281 @@
+
+    .PARAMETER EnableException
+        By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+        This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .PARAMETER WhatIf
+        If this switch is enabled, no actions are performed but informational messages will be displayed that explain what would happen if the command were to run.
+
+    .PARAMETER Confirm
+        If this switch is enabled, you will be prompted for confirmation before executing any operations that change state.
+
+    .NOTES
+        Tags: Firewall, Network
+        Author: Andreas Jordan (@JordanOrdix), ordix.de
+
+        Website: https://dbatools.io
+        Copyright: (c) 2021 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/New-DbaFirewallRule
+
+    .EXAMPLE
+        PS C:\> New-DbaFirewallRule -SqlInstance SRV1, SRV1\TEST -Auto
+
+        Automatically configures the needed firewall rules for both the default instance and the instance named TEST on SRV1.
+
+    .EXAMPLE
+        PS C:\> New-DbaFirewallRule -SqlInstance SRV1, SRV1\TEST -Auto -Configuration @{ Profile = 'Domain', Group = 'SQL' }
+
+        Automatically configures the needed firewall rules for both the default instance and the instance named TEST on SRV1,
+        but configures the firewall rule for the domain profile only and uses the group name 'SQL' instead of the default 'SQL Server'.
+
+    .EXAMPLE
+        PS C:\> $fwConf = @{
+        >>     DisplayName = 'SQL Server'
+        >>     Enabled     = 'True'
+        >>     Direction   = 'Inbound'
+        >>     Protocol    = 'TCP'
+        >>     LocalPort   = 14331
+        >> }
+        PS C:\> New-DbaFirewallRule -SqlInstance SRV2\DEMO -Configuration $fwConf
+
+        Creates a firewall rule with the given configuration for the DEMO instance on SRV2.
+        As -Auto is not used, the command only creates a rule for the instance and no rule for the SQL Server Browser.
+
+    #>
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = "High")]
+    param (
+        [parameter(Mandatory, ValueFromPipeline)]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$Credential,
+        [switch]$Auto,
+        [hashtable]$Configuration,
+        [switch]$EnableException
+    )
+
+    begin {
+        $cmdScriptBlock = {
+            # This scriptblock will be processed by Invoke-Command2.
+            $firewallRuleParameters = $args[0]
+
+            try {
+                $successful = $true
+                $cimInstance = New-NetFirewallRule @firewallRuleParameters -WarningVariable warn -ErrorVariable err -WarningAction SilentlyContinue -ErrorAction SilentlyContinue
```

Comment:
Since this command may not exist, please use this before `if (-not (Get-Command New-NetFirewallRule)) { throw whatever`

Reply:
This is the output, if I change `New-NetFirewallRule` to `New-NetFirewallRuleX`:
![image](https://user-images.githubusercontent.com/66946165/126059016-c5ce755a-1286-48ab-a4dd-04460464ab53.png)

I think that is ok. Do you want to display a different message?

Reply:
Yes, please. Let them know that the command is only supported in XYZ and up, otherwise, it's likely they will believe something is missing in dbatools.